docs/suppressing-instrumentation.md

```diff
@@ -6,11 +6,152 @@ You can disable the agent using `-Dotel.javaagent.enabled=false`
 ## Suppressing specific agent instrumentation
 
 You can suppress agent instrumentation of specific libraries by using
-`-Dotel.instrumentation.[id].enabled=false`.
+`-Dotel.instrumentation.[name].enabled=false` where `name` is the instrumentation `name`
+(main or additional):
 
-where `id` is the instrumentation `id`:
+| Main instrumentation name | Additional names |
+|---------------------------|------------------|
+| akka-actor|akka-actor-2.5 |
+| akka-http|akka-http-client |
+| akka-http|akka-http-server |
+| apache-camel|apache-camel-2.20 |
+| apache-dubbo|apache-dubbo-2.7 |
+| apache-httpasyncclient|apache-httpasyncclient-4.1 |
+| apache-httpclient|apache-httpclient-2.0 |
+| apache-httpclient|apache-httpclient-4.0 |
+| apache-httpclient|apache-httpclient-5.0 |
+| armeria|armeria-1.3 |
+| async-http-client|async-http-client-1.9 |
+| async-http-client|async-http-client-2.0 |
+| aws-lambda|aws-lambda-1.0 |
+| aws-sdk|aws-sdk-1.11 |
+| aws-sdk|aws-sdk-2.2 |
+| axis2|axis2-1.6 |
+| cassandra|cassandra-3.0 |
+| cassandra|cassandra-4.0 |
+| classloader |
+| couchbase|couchbase-2.0 |
+| couchbase|couchbase-2.6 |
+| couchbase|couchbase-3.1 |
```

Comment:
I'd suggest that we swap these, so that the "Main instrumentation name" is unique and matches the instrumentation module name, while the additional names have the shortcuts / version-agnostic names

Reply:
I followed what we have in the code (main vs additional). It would definitely make sense to have the main as unique, but probably better to change it in the code and then reflect in the docs. What do you think?

Reply:
👍

```
public InstrumentationModule(
      String mainInstrumentationName, String... additionalInstrumentationNames) {
```

==>

```
public InstrumentationModule(String moduleName, String... additionalNames) {
```

where `moduleName` is the same as the gradle module name except when there are multiple InstrumentationModules in a single gradle module

Reply:
> where `moduleName` is the same as the gradle module name except when there are multiple InstrumentationModules in a single gradle module

Hmm, if we follow this won't we end up with main module names like `couchbase-2.0`? In that case whenever somebody wants to turn off a particular instrumentation they'll need to know 1) what version of the instrumented library they're using, and 2) what instrumentations it maps to (taking couchbase as an example: version 2.8 actually uses both 2.0 and 2.6 instrumentations).
I think I'd rather get rid of those versioned module names completely (with several exceptions of course; for example the `couchbase-2.6` module could use names `"couchbase", "couchbase-networking"`, with every other couchbase module using just `"couchbase"`)

Reply:
That's a very interesting discussion (unique vs generic) but I think we should move it to a separate issue. Naming is important from both disabling and usage (checking in logs what actually happens) points of view. Deciding this here (documentation task) would not seem the right thing to do :)

Reply:
I asked about this in the SIG meeting yesterday with @iNikem and @anuraaga, and consensus was to list the unversioned module names in this suppressing doc, and we can mention in troubleshooting about disabling a specific instrumentation module using the full module name, but that will just be a troubleshooting feature (e.g. for troubleshooting if muzzle is not doing its job) and we won't list all those versioned names in that doc. (and I think this means we don't need to change anything in the code)

Reply:
Updated :)